lib/addons/core-addon/index.ts

```diff
@@ -39,8 +39,9 @@ export class CoreAddOn implements ClusterAddOn {
         let serviceAccountRoleArn: string | undefined = undefined;
 
         if (this.coreAddOnProps?.policyDocument) {
+            const policyDoc = PolicyDocument.fromJson(this.coreAddOnProps.policyDocument(clusterInfo.cluster.stack.partition));
```

Comment:
If this is the only way to use the policy document, let's make the policyProvider function return PolicyDocument as opposed to some json.